internal/command/action.go

```diff
@@ -22,18 +24,24 @@ type ActionConfig struct {
 	Ignores     []string
 	IgnoreAll   bool
 	Environment string
+	VersionProvider bool		//<---------------------------
 }
 
 const ENV_FILE_MAGIC = "@SUMMONENVFILE"
 const SUMMON_ENV_KEY_NAME = "SUMMON_ENV"
 
 var Action = func(c *cli.Context) {
-	if !c.Args().Present() {
+	if !c.Args().Present() && !c.Bool("version-providers") { //<---------------------------THIS GETS CALLED BEFORE TEST FOR VERSION-PROVIDERS TAG
 		fmt.Println("Enter a subprocess to run!")
 		os.Exit(127)
 	}
 
-	provider, err := prov.Resolve(c.String("provider"))
+	if(c.Bool("version-providers")){ //<----------------- THIS CAN DEFINITELY BE DONE BETTER
+		printProviderVersions()
+		os.Exit(0)
+	}
```

Comment:
This is fine though you probably don't need to explicitly exit - I think you can just `return 0, nil` maybe?

Reply:
Trying to `return 0, nil` causes problems farther down. All of the other parts of that same function use `os.Exit([VALUE])`

Reply:
I guess you can ignore that - after expanding this method to see more of the code, you're right that this doesn't really use the return value (which is silly but it is what it is right now).